test/kubemark/start-kubemark.sh

```diff
@@ -19,10 +19,27 @@
 TMP_ROOT="$(dirname "${BASH_SOURCE}")/../.."
 KUBE_ROOT=$(readlink -e ${TMP_ROOT} 2> /dev/null || perl -MCwd -e 'print Cwd::abs_path shift' ${TMP_ROOT})
 
-source "${KUBE_ROOT}/test/kubemark/common.sh"
 source "${KUBE_ROOT}/test/kubemark/skeleton/util.sh"
 source "${KUBE_ROOT}/test/kubemark/cloud-provider-config.sh"
 source "${KUBE_ROOT}/test/kubemark/${CLOUD_PROVIDER}/util.sh"
+source "${KUBE_ROOT}/cluster/kubemark/${CLOUD_PROVIDER}/config-default.sh"
+source "${KUBE_ROOT}/cluster/kubemark/util.sh"
```

Comment:
Well believe me, the order of sourcing scripts is a very sensitive issue and after investing a few hours on this, I finally got it working. :)
It's a long answer to your question, but here's a crisp reply:
- We need to source `cluster/kubemark/${CLOUD_PROVIDER}/config-default.sh` **after** `cluster/kubemark/util.sh`, as `cluster/kubemark/util.sh` calls gce's config-default.sh somewhere inside which sets MASTER_NAME to "kubernetes-master", while we want it to be set as "kubernetes-kubemark-master"
- We need to also source `cluster/kubemark/${CLOUD_PROVIDER}/config-default.sh` **before** `cluster/kubemark/util.sh` otherwise some variables would be set by `cluster/kubemark/util.sh` that **will not be** over-written by `cluster/kubemark/${CLOUD_PROVIDER}/config-default.sh` due to assignments of the form VAR="${VAR:-"foobar"}".
- And regarding the common scripts that you've mentioned, the only common script called (at the top-level) by run-e2e-tests.sh and start-kubemark.sh is `cluster/kubemark/util.sh`, which should be fine. On the other hand, scripts `test/kubemark/skeleton/util.sh` and `test/kubemark/${CLOUD_PROVIDER}/util.sh` (which overwrites the skeleton/util.sh) should only be called in start- and stop-kubemark scripts as those are the only places where we need those functions.

Hope that answers some questions about why I did stuff the way it is done :)